content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java

```diff
@@ -2098,8 +2098,9 @@ private synchronized void doDragDropUpload(HttpServletRequest request, HttpServl
 					collectionId = resourceGroup;
 				}
 				try{
-					//if user has chosen to overwrite existing resource save the new copy
-					if(overwrite != null && overwrite.equals("true")){
+					//if user has chosen to overwrite existing resource and has correct permissions, save the new copy
+					boolean resourceExists = ContentHostingService.getResource(resourceId)!=null;
```

Comment:
This is slightly misleading code as it will `resourceExists` will never be false as an exception will be thrown instead.